rust/lance-encoding/src/decoder.rs

```diff
@@ -1437,7 +1438,7 @@ impl BatchDecodeStream {
                 let emitted_batch_size_warning = slf.emitted_batch_size_warning.clone();
                 let task = async move {
                     let next_task = next_task?;
-                    next_task.into_batch(emitted_batch_size_warning)
+                    spawn_cpu(move || next_task.into_batch(emitted_batch_size_warning)).await
```

Comment:
Decode work like `fsst` happens here and can block our Tokio worker thread. I believe it's part of the reason we observe unexpected delays in IO scheduling.

Reply:
> This code should be run on the decode path (CPU threads) already? Why do we need to spawn yet another task?

From my profile, I saw that the heavy decoding happens on Tokio’s worker threads instead of our CPU threads.